src/parameterized_tendencies/radiation/radiation.jl

```diff
@@ -32,14 +32,21 @@ function radiation_model_cache(
     Y,
     radiation_mode::RRTMGPI.AbstractRRTMGPMode,
     params,
-    ᶜp; # Used for ozone
+    ᶜp, # Used for ozone
+    aerosol_names;
     interpolation = RRTMGPI.BestFit(),
     bottom_extrapolation = RRTMGPI.SameAsInterpolation(),
     data_loader = rrtmgp_data_loader,
 )
     context = ClimaComms.context(axes(Y.c))
     device = context.device
     (; idealized_h2o, idealized_insolation, idealized_clouds) = radiation_mode
+    if !(radiation_mode isa RRTMGPI.GrayRadiation)
+        (; aerosol_radiation) = radiation_mode
+        if aerosol_radiation
+            @assert "SO4" in aerosol_names
+        end
+    end
```

Comment:
```suggestion
    (; aerosol_radiation) = radiation_mode
    if !(radiation_mode isa RRTMGPI.GrayRadiation) &&
       aerosol_radiation &&
       !("SO4" in aerosol_names)
        error("Aerosol radiation is turned on but SO4 is not provided")
    end
```
I think a more informative error would be helpful, here is what the snippet could look like

Reply:
Thanks. I made the change except for moving (; aerosol_radiation) = radiation_mode in the if block, because GrayRadiation mode doesn't contain aerosol_radiation as an argument.